### PR TITLE
Feature: Added support for canceling tab tear off

### LINKED ISSUE #10234: Feature: Esc to cancel tab tear off
### What feature or improvement do you think would benefit Files?

Sometimes when using a touchpad, I accidentally drag a tab out to a new window. While still holding down the left button, I tried to cancel it with Esc but didn't work. A cancel function would be helpful to quickly prevent unintentional dragging.

### Requirements

- Implement a cancel shortcut when tearing a tab out of the window <kbd>Esc</kbd>

### Files Version

2.3.49.0

### Windows Version

10.0.22623.746

### Comments

_No response_

### PR DESCRIPTION
**Pull request by yaira2**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #10234...

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Open multiple tabs
   2. Drag tab outside the window and release mouse
   3. Confirm tab was teared off
   4. Drag tab outside window and press escape
   5. Confirm the drag operation is canceled

**Screenshots (optional)**
Add screenshots here.

### CODE REVIEW

**Comment by yaira2** [comment]
There seems to be an issue where this doesn't always work.

**Comment by yaira2** [comment]
I'm wondering if there is a focus issue where the focus isn't always on the tag that's being dragged. If this is the case, it can cause the KeyDown event to not be fired.

**Comment by yaira2** [comment]
@gave92 can you try this out?

**Comment by gave92** [comment]
Sry can't test it rn :(

* if you go with this version remember to unsubscribe the event, currently in registers multiple times.

File: src/Files.App/UserControls/TabBar/TabBar.xaml.cs [hunk 1]
```diff
@@ -139,12 +142,24 @@ private void TabHoverSelected(object sender, object e)
...
 			var tabViewItemArgs = (args.Item as TabBarItem).NavigationParameter;
 			args.Data.Properties.Add(TabPathIdentifier, tabViewItemArgs.Serialize());
 			args.Data.RequestedOperation = DataPackageOperation.Move;
 
 			InteropHelpers.GetCursorPos(out dragStartPoint);
 			dragStartTime = DateTimeOffset.UtcNow;
+
+			// Force focus in order that the keydown event works
+			Focus(FocusState.Programmatic);
+			PreviewKeyDown += (s, e) =>
```
Comment (yaira2) [inline src/Files.App/UserControls/TabBar/TabBar.xaml.cs:155]: Is there a way to unsubscribe?

Comment (yaira2) [reply src/Files.App/UserControls/TabBar/TabBar.xaml.cs:155]: Alternatively we can create a separate method, but we'd need a way to access `args.Cancel`.

**Comment by gave92** [comment]
This mostly works, it fails consistently only when pressing ESC while tab is over a second instance of Files.

**Comment by yaira2** [comment]
> This mostly works, it fails consistently only when pressing ESC while tab is over a second instance of Files

The other window is probably taking the focus, we could work around this by increasing the scope of the key press event but I think it's okay as is.

> What issues did you experience with https://github.com/files-community/Files/commit/b92feedb3c694e4b386d0d8bd532e9430b343c3a?

It would cancel the drag even without pressing esc.

**Pull request merged**